### DISCUSSION TITLE
Autocompletion in jsonnet imported libraries with qbec

### DISCUSSION THREAD

**Discussion by rgershkovich** [splunk/qbec#285] [General]
Hello guys,

First of all, thanks a lot for your work on qbec, we hugely depend on this tool.
I understand this is not a qbec-specific question, but as you probably use jsonnet for Kubernetes a lot yourself and also have some suggestions for VSCode [here](https://qbec.io/userguide/usage/tips-and-tricks/) I might give a try.

May I ask you for hints regarding IDE setup for autocompletion from imported libraries? I can't understand if I am doing it wrong, or it is not possible to do currently.

What I want to have (using https://github.com/jsonnet-libs/k8s-libsonnet):
```jsonnet
local k = import '1.23/main.libsonnet';

local container = k.core.v1.container;

container.new(config.name, image) +
container.withVolumeMounts($._config.volume_mounts) +
container.with ...
```
When i write `container.with` I want to have a suggestion from the list of possible values. Would be nice to have suggestion from `k.core.v1 ...` list, but this is less desirable.

What I tried:
* VSCode with https://github.com/grafana/jsonnet-language-server: have to disable tanka mode, does autocomplete for standard library and self/$ but nothing from imported libraries, also you can't specify external jsonnet interpreter, so custom things like `glob-import` are not supported
* VSCode with https://github.com/Sebbia/vscode-jsonnet-ng: nearly same experience, e.g. you can specify `jsonnet-qbec`, but config options are not compatible (like `-J` instead of `--jpath`) and can't be changed, in default mode does not autocomplete from libs
* IntelliJ with https://plugins.jetbrains.com/plugin/18752-jsonnet-language-server: crashes PyCharm CE when enabled with `Could not initialize class com.github.zzehring.intellijjsonnet.JsonnetFileType`

Would be grateful for any recommendations as greping through mixin stuff is suboptimal. Thanks a lot!

**Comment by gotwarlost** [comment]
We don;t actually use a lot of complex libraries in jsonnet. That said...

Would it be helpful if `jsonnet-qbec` supported the same flags as `jsonnet`?
